### PR TITLE
New command: kustomize edit add base

### LINKED ISSUE #12: Support `kustomize edit add base`
Add `kustomize edit add base` command to modify the kustomization.yaml with a base.

was https://github.com/kubernetes/kubectl/issues/481

### PR DESCRIPTION
**Pull request by guineveresaenger**
Fixes #12 

This pull request adds support for editing the kustomization.yaml in the current directory with a base.

### CODE REVIEW

**Comment by Liujingfang1** [comment]
@guineveresaenger Can you squash the commits?

**Comment by guineveresaenger** [comment]
@Liujingfang1 done!

For future reference, what is the gold standard in k8s-sigs for squashing commits? Before asking for a review, or after approval?

File: pkg/commands/addbase.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```
Comment (Liujingfang1) [inline pkg/commands/addbase.go]: change `2017` to `2018` please.

File: pkg/commands/addbase.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+// newCmdAddBase adds the file path of the kustomize base to the kustomization file.
+func newCmdAddBase(fsys fs.FileSystem) *cobra.Command {
+	var o addBaseOptions
+
+	cmd := &cobra.Command{
+		Use:   "base",
+		Short: "Adds a directory path to a base kustomization to the current directory's kustomization file.",
+		Example: `
+		add base {filepath}`,
```
Comment (Liujingfang1) [inline pkg/commands/addbase.go]: How about `add base {filepath1},{filepath2}`? 
Then you can split the argument to get multiple paths.

File: pkg/commands/addbase_test.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```
Comment (Liujingfang1) [inline pkg/commands/addbase_test.go]: change `2017` to `2018` please

**Comment by Liujingfang1** [comment]
> For future reference, what is the gold standard in k8s-sigs for squashing commits? Before asking for a review, or after approval?

For the change focused on the same scope and with a reasonable size, you can squash the commits into one before a review. For some change, you may want to split them into different commits with clear logic boundary. For example, say you need to update the dependency in vendor directory for `add base` command. You can put updating vendor in one commit and put other code changes in another commit.

**Comment by Liujingfang1** [comment]
The failure is from the unit test you added `TestAddBaseHappyPath`. You can run it locally and fix it.

**Comment by guineveresaenger** [comment]
So... when I test the entire commands package I get a test (and a lot of related sub-tests and output) that says --- FAIL: TestDiff (0.12s), which makes local test runs difficult to debug.
That output exists on master too, for me. First few lines of failed test output:
```
--- FAIL: TestDiff (0.15s)
    --- FAIL: TestDiff/base-only (0.01s)
    	diff_test.go:124: diff -u -N /var/folders/db/87sjkx2d5z98_l7wj78tk2jw0000gn/T/noop-753565031/apps_v1beta2_Deployment_nginx.yaml /var/folders/db/87sjkx2d5z98_l7wj78tk2jw0000gn/T/transformed-089911336/apps_v1beta2_Deployment_nginx.yaml
    		--- /var/folders/db/87sjkx2d5z98_l7wj78tk2jw0000gn/T/noop-753565031/apps_v1beta2_Deployment_nginx.yaml	YYYY-MM-DD HH:MM:SS
    		+++ /var/folders/db/87sjkx2d5z98_l7wj78tk2jw0000gn/T/transformed-089911336/apps_v1beta2_Deployment_nginx.yaml	YYYY-MM-DD HH:MM:SS
    		@@ -1,14 +1,27 @@
    		 apiVersion: apps/v1beta2
    		 kind: Deployment
    		 metadata:
    		+  annotations:
    		+    note: This is a test annotation
```

File: pkg/commands/addbase.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	baseDirectoryPaths string
+}
+
+// newCmdAddBase adds the file path of the kustomize base to the kustomization file.
+func newCmdAddBase(fsys fs.FileSystem) *cobra.Command {
+	var o addBaseOptions
+
+	cmd := &cobra.Command{
+		Use:   "base",
+		Short: "Adds a directory path to a base kustomization to the current directory's kustomization file.",
```
[the remark this answers is not in the document]
Comment (guineveresaenger) [reply pkg/commands/addbase.go]: Fixed!

**Pull request merged**